pkg/api/handlers/compat/images_build.go

```diff
@@ -224,8 +224,13 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 				// it's not json, assume just a string
 				m = []string{query.Dockerfile}
 			}
+
 			for _, containerfile := range m {
-				containerFiles = append(containerFiles, filepath.Join(contextDirectory, filepath.Clean(filepath.FromSlash(containerfile))))
+				if containerfile[0] == byte('/') {
```

Comment:
Shouldn't this just check if containerfile is a valid URL using `ParseRequestURI` and treat every other value as a path ? I mean could value be a relative path which does not starts with `/`

Reply:
A path is treated as a valid URI. Perhaps I can do the same check as before if the parsed requests path equals the path, then it is a path.